src/main/java/com/kryptnostic/datastore/services/EdmService.java

```diff
@@ -694,12 +710,15 @@ private void removePropertyTypesFromEntityType(
          * property types are checked to exist.
          */
         if ( isValid ) {
-            if ( properties != null && entityType.getProperties() != null ) {
+
+            if ( properties != null && entityType.getProperties() != null && entityType.getProperties().containsAll( properties ) ) {
                 entityType.removeProperties( properties );
                 // Acl
                 properties
                         .forEach( propertyTypeFqn -> permissionsService.removePermissionsForPropertyTypeInEntityType(
                                 entityType.getFullQualifiedName(), propertyTypeFqn ) );
+            } else {
+                throw new BadRequestException( "Not all properties are included in the EntityType" );
```

Comment:
IllegalArgumentException should be thrown; HTTP status codes are returned at Controller level.